### PR TITLE
Fixed tests and CLHEP linking error

### PR DESCRIPTION
**Pull request by whit2333**
This could be due to using a system-wide CLHEP instead of the built-in geant4 version.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -110,6 +110,7 @@ add_subdirectory(DDCond)
 add_subdirectory(DDAlign)
 add_subdirectory(DDEve)
 add_subdirectory(DDDB)
+add_subdirectory(DDTest)
 
 dd4hep_enable_tests( DDTest )
```
Comment (andresailer) [inline CMakeLists.txt]: DDTest is added if BUILD_TESTING=ON, this is not the proper solution for your problem. There is an issue in UtilityApps/CMakeLists.txt where DDTest is required for one executable, but then things fail.

Comment (whit2333) [inline CMakeLists.txt]: Actually build_testing=off fails too. See this issue
https://github.com/AIDASoft/DD4hep/issues/80

**Comment by whit2333** [comment]
This build fails because DDRec needs to be passed CLHEP when geant4 is built with a system-wide CLHEP. This test uses the builtin geant4.

**Comment by petricm** [comment]
The Geant in the CI is build with a sytem CLHEP (GEANT4_USE_SYSTEM_CLHEP=ON and CLHEP_ROOT_DIR=__path__)

**Comment by whit2333** [comment]
Looking at the build 
```

CMake Warning at cmake/DD4hepBuild.cmake:395 (find_package):

  By not providing "FindCLHEP.cmake" in CMAKE_MODULE_PATH this project has

  asked CMake to find a package configuration file provided by "CLHEP", but

  CMake did not find one.

```

**Comment by whit2333** [comment]
I don't follow? Why is it failing to find CLHEP ?

**Comment by whit2333** [comment]
Looking at the travis CI output:

`CMake Warning at cmake/DD4hepBuild.cmake:395 (find_package):

  By not providing "FindCLHEP.cmake" in CMAKE_MODULE_PATH this project has

  asked CMake to find a package configuration file provided by "CLHEP", but

  CMake did not find one.

  Could not find a package configuration file provided by "CLHEP" with any of

  the following names:

    CLHEPConfig.cmake

    clhep-config.cmake

  Add the installation prefix of "CLHEP" to CMAKE_PREFIX_PATH or set

  "CLHEP_DIR" to a directory containing one of the above files.  If "CLHEP"

  provides a separate development package or SDK, be sure it has been

  installed.

Call Stack (most recent call first):

  cmake/DD4hepBuild.cmake:486 (dd4hep_find_packageEx)

  cmake/DD4hepBuild.cmake:863 (dd4hep_find_package)

  DDRec/CMakeLists.txt:14 (dd4hep_package)

CMake Error at cmake/DD4hepBuild.cmake:185 (message):

  ++> [DDRec] Unknown package CLHEP

Call Stack (most recent call first):

  cmake/DD4hepBuild.cmake:415 (dd4hep_fatal)

  cmake/DD4hepBuild.cmake:486 (dd4hep_find_packageEx)

  cmake/DD4hepBuild.cmake:863 (dd4hep_find_package)

  DDRec/CMakeLists.txt:14 (dd4hep_package)

-- Configuring incomplete, errors occurred!

See also "/DD4hep/build/CMakeFiles/CMakeOutput.log".

See also "/DD4hep/build/CMakeFiles/CMakeError.log".`

**Comment by petricm** [comment]
You can try moving the add_direcotry after  the inclusion of Geant, this might do it. But the real problem is what @andresailer said.

**Comment by petricm** [comment]
This is not good, find_package for CLHEP should only be used if you are building without Geant, since otherwise the Geant4_USE_FILE tells cmake which and where to get CLHEP.

**Comment by whit2333** [comment]
OK I figured out what the problems where.

1. BUILD_TESTING currently isn't an option at all. It is essentially required.

2. A link to CLHEP is leaking into DDDetectors which is configured before "FIND_PACKAGE(Geant4...)". Therefore linking fails at DDDetectors. Instead if finding out why DDDetectors was linking against CLHEP, I just moved laoding the the geant4 package before adding the subdirectories.

**Pull request closed without merging**